pkg/cri/server/images/image_pull.go

```diff
@@ -332,23 +312,14 @@ func (c *CRIImageService) UpdateImage(ctx context.Context, r string) error {
 		return fmt.Errorf("get image by reference: %w", err)
 	}
 	if err == nil && img.Labels()[crilabels.ImageLabelKey] != crilabels.ImageLabelValue {
-		// Make sure the image has the image id as its unique
-		// identifier that references the image in its lifetime.
-		configDesc, err := img.Config(ctx)
+		namedRef, err := distribution.ParseDockerRef(r)
 		if err != nil {
-			return fmt.Errorf("get image id: %w", err)
-		}
-		id := configDesc.Digest.String()
-		labels := c.getLabels(ctx, id)
-		if err := c.createImageReference(ctx, id, img.Target(), labels); err != nil {
-			return fmt.Errorf("create image id reference %q: %w", id, err)
+			return err
 		}
-		if err := c.imageStore.Update(ctx, id); err != nil {
-			return fmt.Errorf("update image store for %q: %w", id, err)
-		}
-		// The image id is ready, add the label to mark the image as managed.
-		if err := c.createImageReference(ctx, r, img.Target(), labels); err != nil {
-			return fmt.Errorf("create managed label: %w", err)
+		labels := c.getLabels(ctx, r)
+		_, err = c.updateImageReferences(ctx, namedRef, img, labels, false)
```

Comment:
are we ignoring the isSchemaV1 flag here, so repoDigest might be different for schema v1 images? 

I am ok with this since we are deprecating schema v1 in 2.0 (https://containerd.io/releases/#deprecated-features), just double checking.

Reply:
IIUC schemav1 digest is converted when the image is pulled. so I don't think it's a problem.

https://github.com/containerd/containerd/blob/261e01c2acc1eb03ac15bda80c9230a0d0dcc722/client/pull.go#L199-L211
https://github.com/containerd/containerd/blob/261e01c2acc1eb03ac15bda80c9230a0d0dcc722/client/pull.go#L271-L275